specification/refarch.adoc

```diff
@@ -147,311 +189,326 @@ assigned to the TVM by the VMM.
+with reason) to initiate the return of execution control to the OS/VMM if
+required. The TSM-driver restores the context for the OS/VMM via the
 per-hart control sub-structure THCS.hssa (See <<appendix_a>>).This canonical
 flow is shown in figure 3.
 
-Beyond the basic operation described above, the following different 
+Beyond the basic operation described above, the following different
 operational models of the TSM may be supported by an implementation:
 
-* *Uninterruptible* *TSM* - In this model, the TSM security routines are 
-executed in an uninterruptible manner for S-mode interrupts (M-mode 
-interrupts are not inhibited). This implies that the TSM execution always 
-starts from a fixed initial state of the TSM harts and completes the 
-execution with either a TEERET to return control to the OS/VMM or via an 
+* *Uninterruptible* *TSM* - In this model, the TSM security routines are
+executed in an uninterruptible manner for S-mode interrupts (M-mode
+interrupts are not inhibited). This implies that the TSM execution always
+starts from a fixed initial state of the TSM harts and completes the
+execution with either a TEERET to return control to the OS/VMM or via an
 SRET to enter into a TVM (where the execution may be interruptible again).
 
-* *Interruptible TSM with no re-entrancy* - In this model, after the 
-initial entry to the TSM with S-mode interrupts disabled, the TSM enables 
-interrupts during execution of the TSM security routines. The TSM may 
-install its interrupt handlers at this entry (or may be installed via the 
-TEECALL flow as shown below). On an S-mode interrupt, the TSM hart context 
-is saved by the TSM and keeps the interrupt pending. The TSM may then 
-TEERET to the host OS/VMM with explicit information about the interruption 
-provided via the pending interrupt to the OS/VMM. The TSM-driver supports a 
-TEERESUME ECALL which enables the TSM to enforce that the resumption of the 
-interrupted TSM security routine is initiated by the OS/VMM on the same 
-hart. The TSM hart context restore is enforced by the TSM to allow for the 
-resumed TSM security routine operation to complete. An example of an 
-interruptible flow is the conversion of a large 2MB page to confidential 
-memory, which may require a long latency encryption operation. Intermediate 
-state of the operation must be saved and restored by the TSM for such 
-flows. 
-
-**__This specification describes the operation of the TSM in this 
+* *Interruptible TSM with no re-entrancy* - In this model, after the
+initial entry to the TSM with S-mode interrupts disabled, the TSM enables
+interrupts during execution of the TSM security routines. The TSM may
+install its interrupt handlers at this entry (or may be installed via the
+TEECALL flow as shown below). On an S-mode interrupt, the TSM hart context
+is saved by the TSM and keeps the interrupt pending. The TSM may then
+TEERET to the host OS/VMM with explicit information about the interruption
```

Comment:
unless the interrupt is to be handled by the TSM itself?

Reply:
don't think we want to require any cases where the TSM handles the interrupts unilaterally.

Reply:
How about RAS and Timer interrupts?

Reply:
TSM can take an action and forward the event to the host VMM